pkg/volume/util/fsquota/quota_linux.go

```diff
@@ -439,3 +445,24 @@ func ClearQuota(m mount.Interface, path string) error {
 	}
 	return nil
 }
+
+// RecordProjectIDOnProjectFiles -- Records project file for the given directory-path to /etc/projects and /etc/projectid
+func RecordProjectIDOnProjectFiles(podVolumeDir string) error {
```

Comment:
This needs to be part of the interface, and a null version implemented in `quota.go`.  Otherwise it won't build on non-Linux systems.

Also make sure that this works with persistent /etc.

Reply:
> This needs to be part of the interface, and a null version implemented in quota.go. Otherwise it won't build on non-Linux systems.

Done
> Also make sure that this works with persistent /etc.

Added a check for it. Want to know your thoughts on it: https://github.com/kubernetes/kubernetes/pull/82537/files#diff-1556842a9083b228aa95143da7468b79R468
If we need to write just in the file, we probably cannot use the `createProjectID` method and might instead need to use `writeprojectfile` method

Reply:
Then perhaps `createProjectID` needs to be refactored.

Reply:
How do you feel about using the `writeProjectFile` method instead?